src/components/FormType.tsx

```diff
@@ -34,37 +35,88 @@ const MailFormType = () => <div>this is a mail only form</div>;
 const CalendarFormType = () => <div>this is a calendar only form</div>;
 const GitHubFormType = ({ modals }: GitHubFormTypeProps) => {
   const { setCreateModalState, setRulesetModalState } = modals;
+  const [loading, setLoading] = useState(false);
   const { data: session } = useSession();
 
   const createWebhook = api.webhook.createWebhook.useMutation();
   const createAutomation = api.automation.createAutomation.useMutation();
+  const userAutomations = api.automation.getUserAutomations.useQuery();
 
   const handleSubmit = async (e: React.FormEvent) => {
-    if (!session) return "log out and log back in";
     e.preventDefault();
+    setLoading(true);
+    if (loading) {
+      toast.dismiss();
+      toast.error("Stop spamming 😡");
+    } else {
+      if (!session) {
+        toast.dismiss();
+        toast.error("Log Out and Log Back In");
+        setLoading(false);
+        return;
+      }
+      toast.dismiss();
+      toast.loading("Reviewing your conditions and permissions...");
 
-    // passing accessToken is necessary because DB access token can be stale which can cause 401: Bad Credentials
-    const createWebhookResult = await createWebhook.mutateAsync({
-      accessToken: session.user.accessToken,
-    });
-    if (createWebhookResult === false) {
-      console.log("log out and log back in");
-      return;
-    } else if (createWebhookResult === 422) {
-      console.log("hook already exists");
-      return;
-    }
+      // passing accessToken is necessary because DB access token can be stale which can cause 401: Bad Credentials
+      const createWebhookResult = await createWebhook.mutateAsync({
+        accessToken: session.user.accessToken,
+      });
+      if (createWebhookResult === false) {
+        toast.dismiss();
+        toast.error("Log Out and Log Back In");
```

Comment:
can you explain why "log out and log back in" fixes webhook creation?

is thie related to the question you asked in the group/has that been resolved?

Reply:
> is thie related to the question you asked in the group/has that been resolved?

Yeah, this is the same question from the group that was asked by me. It hasn't been resolved. I've responded to a very similar question about why it is needed for the user to logout. This was said in the first part of the response by me:

"For the current implementation, if there is a situation where the session is undefined (can possibly occur when the access token in the DB is stale and user is in the page for a good amount of time and refreshes). This occurs that the access token can't be updates since the session is undefined and it's required for the user to log out and log back in. When the user logs out and logs back in, they don't need to manually enter their password."

Reply:
So you're using the logout/login as a way to manually refresh the token?

Reply:
Yes. Since access tokens are short lived. Something like this can occur but not very often.